### ISSUE TITLE
Depreciation Warning and Error

### ISSUE THREAD

**Issue by Nick425321** [PeteRager/lennoxs30#338]
I noticed today one of my S30 zones was set incorrectly and after poking around  I found these error/warning messages in the log, the warning I understand, but the error I cannot figure it out. Please let me know if you need any other info/log.

Here is the script/action:

alias: iComfort Close Openings No Guests Mode
sequence:
  - if:
      - condition: device
        device_id: 788e9a8df376b12ec1459495142f2e5f
        domain: climate
        entity_id: 19bdfa3fa4eb5510fb73872f9bcf35a3
        type: is_preset_mode
        preset_mode: Windows Open
    then:
      - alias: "Change Mode Cancel Windows Open "
        device_id: 788e9a8df376b12ec1459495142f2e5f
        domain: climate
        entity_id: 19bdfa3fa4eb5510fb73872f9bcf35a3
        type: set_preset_mode
        preset_mode: schedule IQ
  - delay:
      hours: 0
      minutes: 0
      seconds: 0
      milliseconds: 800
    enabled: true
  - device_id: 788e9a8df376b12ec1459495142f2e5f
    domain: climate
    entity_id: 19bdfa3fa4eb5510fb73872f9bcf35a3
    type: set_preset_mode
    preset_mode: schedule IQ
  - delay:
      hours: 0
      minutes: 0
      seconds: 0
      milliseconds: 800
    enabled: true
  - device_id: 39911a62cd0a0e8f902c47f4a91ec230
    domain: climate
    entity_id: 7537c311ff9124fcc2d37428b2c9cc87
    type: set_preset_mode
    preset_mode: Z2 Master Bedroom
  - delay:
      hours: 0
      minutes: 0
      seconds: 0
      milliseconds: 800
    enabled: true
  - type: turn_off
    device_id: ee9700bf4e835fff252710418c94aa96
    entity_id: 5f4c801f8819504024825ec9a1511755
    domain: switch
  - delay:
      hours: 0
      minutes: 0
      seconds: 0
      milliseconds: 800
    enabled: true
  - device_id: 19078cd81ce62d9a52c9d137d7b2fe13
    domain: climate
    entity_id: 7ee505503ad2c0b146a3f9f7ae5e7144
    type: set_preset_mode
    preset_mode: Z3 Front Rooms
  - delay:
      hours: 0
      minutes: 0
      seconds: 0
      milliseconds: 800
    enabled: true
  - device_id: 788e9a8df376b12ec1459495142f2e5f
    domain: climate
    entity_id: 19bdfa3fa4eb5510fb73872f9bcf35a3
    type: set_preset_mode
    preset_mode: cancel hold
mode: single
icon: mdi:script-text-play-outline
description: ""

Logger: homeassistant.components.script.icomfort_close_openings_no_guests_mode
Source: helpers/script.py:2030
integration: Script (documentation, issues)
First occurred: 14:34:09 (1 occurrences)
Last logged: 14:34:09

iComfort Close Openings No Guests Mode: Error executing script. Error for device at pos 5: Entity climate.icomfort_z2_mastersuite does not support this service.

Logger: homeassistant.helpers.entity
Source: helpers/entity.py:1637
First occurred: 14:30:02 (2 occurrences)
Last logged: 14:30:02

Entity climate.icomfort_z2_mastersuite (<class 'custom_components.lennoxs30.climate.S30Climate'>) is using deprecated supported features values which will be removed in HA Core 2025.1. Instead it should use <ClimateEntityFeature: 0>, please create a bug report at https://github.com/PeteRager/lennoxs30/issues and reference https://developers.home-assistant.io/blog/2023/12/28/support-feature-magic-numbers-deprecation
Entity climate.icomfort_z3_front (<class 'custom_components.lennoxs30.climate.S30Climate'>) is using deprecated supported features values which will be removed in HA Core 2025.1. Instead it should use <ClimateEntityFeature: 0>, please create a bug report at https://github.com/PeteRager/lennoxs30/issues and reference https://developers.home-assistant.io/blog/2023/12/28/support-feature-magic-numbers-deprecation

<html>
<body>

Version | core-2024.8.2
-- | --
Installation type | Home Assistant OS
Development | false
Supervisor | true
Docker | true
User | root
Virtual environment | false
Python version | 3.12.4
Operating system family | Linux
Operating system version | 6.6.31-haos-raspi
CPU architecture | aarch64
Timezone | America/Los_Angeles
Configuration directory | /config

</body>
</html>

**Comment by PeteRager** [comment]
It looks like you are on a very old version of the integration.  What version are you running?

**Comment by Nick425321** [comment]
2024.8.0

**Comment by PeteRager** [comment]
Please attach the diagnostics file for the integration.

**Comment by Nick425321** [comment]
Pete I have attached the log file, I think I have solved the problem, I had:

device_id: 788e9a8df376b12ec1459495142f2e5f
domain: climate
entity_id: 19bdfa3fa4eb5510fb73872f9bcf35a3
type: set_preset_mode
preset_mode: cancel hold
enabled: true

a cancel hold at the end of my script/action from when I first made my
automation, but it appears that it is no longer required. I have only been
using HA for 6 months, so I'm by no means a power user, and have only
recently spent some time looking at warnings and errors and trying to
eliminate/fix them. I have looked at the attached log and nothing looks
wrong to me, also the system log is not showing the error message now I
have disabled the "cancel hold".
